server.go

```diff
@@ -0,0 +1,110 @@
+	"net/http"
+	"strconv"
+
+	"github.com/minio/minio-go"
+	"github.com/minio/minio/pkg/fs"
+	"github.com/minio/minio/pkg/probe"
+)
+
+func configureWebServer(fsPath string) (*http.Server, *probe.Error) {
+	srvConfig, err := loadConfig()
+	if err != nil {
+		return nil, err.Trace()
+	}
+
+	// Split the api address into host and port.
+	host, port, e := net.SplitHostPort(srvConfig.Addr)
+	if e != nil {
+		return nil, probe.NewError(e)
+	}
+	webPort, e := strconv.Atoi(port)
+	if e != nil {
+		return nil, probe.NewError(e)
+	}
+
+	// Always choose the next port, based on the API address port.
+	webPort = webPort + 1
+	webAddress := net.JoinHostPort(host, strconv.Itoa(webPort))
+
+	// Default host is 'localhost', if no host present.
+	if host == "" {
+		host = "localhost"
+	}
+
+	// Insecure true when TLS is false.
+	var inSecure bool
+	inSecure = !(srvConfig.TLSConfig.CertFile != "" && srvConfig.TLSConfig.KeyFile != "")
+
+	// Initialize minio client for AWS Signature Version '4'
+	client, e := minio.NewV4(net.JoinHostPort(host, port), srvConfig.Credential.AccessKeyID, srvConfig.Credential.SecretAccessKey, inSecure)
+	if e != nil {
+		return nil, probe.NewError(e)
+	}
+
+	web := &WebAPI{
+		FSPath:          fsPath,
+		Client:          client,
+		inSecure:        inSecure,
+		apiAddress:      webAddress,
+		accessKeyID:     srvConfig.Credential.AccessKeyID,
+		secretAccessKey: srvConfig.Credential.SecretAccessKey,
+	}
+
+	// Get web API handler.
+	webHandler := getWebAPIHandler(web)
+
+	// Minio server config.
+	webServer := &http.Server{
+		Addr:           webAddress,
+		Handler:        webHandler,
+		MaxHeaderBytes: 1 << 20,
+	}
+
+	if !inSecure {
+		var err error
+		webServer.TLSConfig = &tls.Config{}
+		webServer.TLSConfig.Certificates = make([]tls.Certificate, 1)
+		webServer.TLSConfig.Certificates[0], err = tls.LoadX509KeyPair(srvConfig.TLSConfig.CertFile, srvConfig.TLSConfig.KeyFile)
+		if err != nil {
+			return nil, probe.NewError(err)
+		}
+	}
+	return webServer, nil
+}
+
+// configureAPIServer configure a new server instance.
+func configureAPIServer(backend fs.Filesystem) (*http.Server, *probe.Error) {
+	srvConfig, err := loadConfig()
+	if err != nil {
+		return nil, err.Trace()
+	}
+
+	// API Handler with embedded backend filesystem.
+	apiHandler := getCloudStorageAPIHandler(CloudStorageAPI{Filesystem: backend})
+
+	apiServer := &http.Server{}
+
+	// Insecure true when TLS is false.
+	var inSecure bool
+	inSecure = !(srvConfig.TLSConfig.CertFile != "" && srvConfig.TLSConfig.KeyFile != "")
+
+	if !inSecure {
+		var err error
+		apiServer.TLSConfig = &tls.Config{}
+		apiServer.TLSConfig.Certificates = make([]tls.Certificate, 1)
+		apiServer.TLSConfig.Certificates[0], err = tls.LoadX509KeyPair(srvConfig.TLSConfig.CertFile, srvConfig.TLSConfig.KeyFile)
+		if err != nil {
+			return nil, probe.NewError(err)
+		}
+	}
+	// Minio server config
+	apiServer.Addr = srvConfig.Addr
+	apiServer.Handler = apiHandler
+	apiServer.MaxHeaderBytes = 1 << 20
```

Comment:
1MB is the default. You dont need to set it explicitly. Even otherwise, it should always require a comment like // 1 MB